pages/definitions/[slug].tsx

```diff
@@ -1,56 +1,72 @@
 import { GetStaticPaths, GetStaticProps } from 'next';
-import React from 'react';
-import ReactMarkdown from 'react-markdown';
 import Breadcrumb from '#components-ui/breadcrumb';
 import ButtonLink from '#components-ui/button';
 import TextWrapper from '#components-ui/text-wrapper';
 import Meta from '#components/meta';
-import { NextPageWithLayout } from 'pages/_app';
 import {
   IDefinition,
   allDefinitions,
   getDefinition,
-} from '#models/definitions';
+} from '#models/article/definitions';
+import { NextPageWithLayout } from 'pages/_app';
 
-const DefinitionPage: NextPageWithLayout<{ definition: IDefinition }> = ({
-  definition,
-}) => (
-  <>
-    <Meta
-      title={definition.seo.title || definition.title}
-      description={definition.seo.description}
-      noIndex={false}
-      canonical={`https://annuaire-entreprises.data.gouv.fr/definitions/${definition.slug}`}
-    />
-    <TextWrapper>
-      <Breadcrumb
-        links={[
-          { href: '/definitions', label: 'Définitions' },
-          { href: '', label: definition.title },
-        ]}
+const DefinitionPage: NextPageWithLayout<{
+  definition: IDefinition;
+}> = ({ definition }) => {
+  return (
+    <>
+      <Meta
+        title={definition.seo.title || definition.title}
+        description={definition.seo.description}
+        noIndex={false}
+        canonical={`https://annuaire-entreprises.data.gouv.fr/definitions/${definition.slug}`}
       />
-      <h1 className="definition-title">{definition.title}</h1>
-      <ReactMarkdown>{definition.body}</ReactMarkdown>
-      {definition.cta ? (
-        <div className="layout-left">
-          <ButtonLink to={definition.cta.to}>{definition.cta.label}</ButtonLink>
-        </div>
-      ) : null}
-      {definition.more ? (
-        <div>
-          <h2>Sur le même sujet</h2>
+      <TextWrapper>
+        <Breadcrumb
+          links={[
+            { href: '/definitions', label: 'Définitions' },
+            { href: '', label: definition.title },
+          ]}
+        />
+        <h1 className="definition-title">{definition.title}</h1>
+        <nav>
+          <h2>Sommaire</h2>
           <ul>
-            {definition.more.map(({ href, label }) => (
-              <li key={href}>
-                <a href={href}>{label}</a>
-              </li>
-            ))}
+            {definition.body.headings
+              .filter(({ depth }) => depth === 2)
+              .map(({ value, id }) => (
+                <li key={id}>
+                  <a href={`#${id}`}>{value}</a>
+                </li>
+              ))}
           </ul>
-        </div>
-      ) : null}
-    </TextWrapper>
-  </>
-);
+        </nav>
+
+        <div dangerouslySetInnerHTML={{ __html: definition.body.html }} />
```

Comment:
Also worth it to  move it into a `ui-component` as it will replace every former `<ReactMarkdown />`

Reply:
I'm not sure if there is a need for an additional abstraction for this line. Unless you meant something else ?

Reply:
I meant the abstraction as I am not a big fan of `dangerouslySetInnerHTML={{ __html: definition.body.html }}` as it can easily break the whole page if the html is not valid